src/arrayMethodSort.js

```diff
@@ -1,12 +1,78 @@
 'use strict';
 
-/**
- * Implement method Sort
- */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const undefinedArr = [];
+
+    for (let i = 0; i < this.length; ) {
+      if (this[i] === null) {
+        undefinedArr.unshift(...this.splice(i, 1));
+      }
+
+      if (typeof this[i] === 'undefined') {
+        undefinedArr.push(...this.splice(i, 1));
+      } else {
+        this[i] = this[i].toString();
+        i++;
+      }
+    }
+
+    for (let j = 0; j < this.length; j++) {
+      for (let i = 0; i < this.length; i++) {
+        let result;
+
+        if (compareFunction) {
+          result = compareFunction(this[i], this[i + 1]);
+        }
+
+        if (!compareFunction) {
+          result = this[i] > this[i + 1];
+        }
+
+        if (result > 0) {
+          const temp = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = temp;
+        }
+      }
+    }
+
+    for (let i = 0; i < undefinedArr.length; i++) {
+      if (undefinedArr[i] === null || typeof undefinedArr[i] === 'undefined') {
+        this.push(undefinedArr[i]);
+      }
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (Number(this[i])) {
```

Comment:
The condition `Number(this[i])` will evaluate to `false` for `0`, which is a valid number. Consider using `!isNaN(Number(this[i]))` to check if the conversion to a number is valid.